documentation/lesson-plans/W2-lesson-plan.md

```diff
@@ -9,12 +9,21 @@
 - Lifecycle methods are used when render is not enough on its own
 
 - Cover each, giving examples of when they might be useful
-  - componentWillMount: SSR
+  - constructor: initializing state
```

Comment:
A `constructor` is not a lifecycle method. It shouldn't be included here, it will confuse students. As you know, `constructors` are pure JS.

I believe it's specifically mentioned elsewhere in the curriculum I wrote that state can be initialised in two ways:

1. Class properties, eg: `state = { myThing: true` }` inside the class directly
2. The `constructor`, as you say here.

Generally class properties can be used for simple initialisation. Facebook uses class properties in their internal source code - it's not something controversial like decorators.

A `constructor` can be used to initialise state when you need more complicated logic (checking other data stores, cookies, localStorage, etc).